src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -1106,7 +1143,7 @@ public void checkout(Run<?, ?> build, Launcher launcher, FilePath workspace, Tas
             ext.beforeCheckout(this, build, git, listener);
         }
 
-        retrieveChanges(build, git, listener);
+        retrieveChanges(build, git, listener, GitUtils.workspaceToNode(workspace));
```

Comment:
Well, it uses Jenkins.getActiveInstance which is Nonnull... But the 2 tests my patch is failing seem to have an NPE caused by a null result of Jenkins.getInstance (which is Nonnull in the API reference too).